openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -75,7 +75,11 @@ def get_account_settings(request, usernames=None, configuration=None, view=None)
 
     requested_users = User.objects.select_related('profile').filter(username__in=usernames)
     if not requested_users:
-        raise UserNotFound()
+        if usernames[0] == request.user:
+            requested_users = User.objects.filter(username__in=usernames)
```

Comment:
instead of doing a `filter()` here, do `get()` and catch the django model `ObjectDoesNotExist` exception and handle the exception.